controllers/snapshot/snapshot_adapter_test.go

```diff
@@ -0,0 +1,343 @@
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ = Describe("Snapshot Adapter", Ordered, func() {
+	var (
+		adapter *Adapter
+
+		testReleasePlan         *releasev1alpha1.ReleasePlan
+		hasApp                  *appstudiov1alpha1.Application
+		hasComp                 *appstudiov1alpha1.Component
+		hasSnapshot             *appstudioshared.ApplicationSnapshot
+		testpipelineRun         *tektonv1beta1.PipelineRun
+		integrationTestScenario *integrationv1alpha1.IntegrationTestScenario
+		env                     appstudioshared.Environment
+		sample_image            string
+	)
+	const (
+		SampleRepoLink = "https://github.com/devfile-samples/devfile-sample-java-springboot-basic"
+	)
+
+	BeforeAll(func() {
+
+		hasApp = &appstudiov1alpha1.Application{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "application-sample",
+				Namespace: "default",
+			},
+			Spec: appstudiov1alpha1.ApplicationSpec{
+				DisplayName: "application-sample",
+				Description: "This is an example application",
+			},
+		}
+
+		Expect(k8sClient.Create(ctx, hasApp)).Should(Succeed())
+
+		integrationTestScenario = &integrationv1alpha1.IntegrationTestScenario{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "example-pass",
+				Namespace: "default",
+
+				Labels: map[string]string{
+					"test.appstudio.openshift.io/optional": "false",
+				},
+			},
+			Spec: integrationv1alpha1.IntegrationTestScenarioSpec{
+				Application: "application-sample",
+				Bundle:      "quay.io/kpavic/test-bundle:component-pipeline-pass",
+				Pipeline:    "component-pipeline-pass",
+				Environment: integrationv1alpha1.TestEnvironment{
+					Name:   "envname",
+					Type:   "POC",
+					Params: []string{},
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, integrationTestScenario)).Should(Succeed())
+
+		testReleasePlan = &releasev1alpha1.ReleasePlan{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-releaseplan-",
+				Namespace:    "default",
+				Labels: map[string]string{
+					releasev1alpha1.AutoReleaseLabel: "true",
+				},
+			},
+			Spec: releasev1alpha1.ReleasePlanSpec{
+				Application: hasApp.Name,
+				Target: kcp.NamespaceReference{
+					Namespace: "default",
+					Workspace: "workspace-sample",
+				},
+			},
+		}
+		Expect(k8sClient.Create(ctx, testReleasePlan)).Should(Succeed())
+
+		env = appstudioshared.Environment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "envname",
+				Namespace: "default",
+			},
+			Spec: appstudioshared.EnvironmentSpec{
+				Type:               "POC",
+				DisplayName:        "my-environment",
+				DeploymentStrategy: appstudioshared.DeploymentStrategy_Manual,
+				ParentEnvironment:  "",
+				Tags:               []string{},
+				Configuration: appstudioshared.EnvironmentConfiguration{
+					Env: []appstudioshared.EnvVarPair{
+						{
+							Name:  "var_name",
+							Value: "test",
+						},
+					},
+				},
+				//				UnstableConfigurationFields: &appstudioshared.UnstableEnvironmentConfiguration{
```

Comment:
nit: I propose removing this commented section until it's needed by the unit tests.

Reply:
OK. I'll remove it. Thanks!

Reply:
Removed